src/pages/EditMonitor.vue

```diff
@@ -1315,10 +1399,20 @@ message HealthCheckResponse {
                     this.processing = false;
                     this.$root.toastRes(res);
                     this.init();
+
+                    // Start the new parent monitor after edit is done
+                    if (createdNewParent) {
+                        this.startParentGroupMonitor();
+                    }
                 });
             }
         },
 
+        async startParentGroupMonitor() {
+            await sleep(2000);
```

Comment:
I put the logic into add/edit's callback, but actually there is still a race condition if the heartbeat checking is too long. I added 2 seconds to improve the situation. I think it is acceptable as there is no callback for heartbeat.

Reply:
I see, yeah running after `editMonitor` is done should be better. Callbacks are making me confused.